### PR TITLE
Added JUnit tests for provider package

### PR DESCRIPTION
**Pull request by VesnaGrumic**
Added JUnit tests for FuseChannel class
Added JUnit tests for Callbacks class
Added JUnit tests for AbstractMqttChannel class
Added JUnit tests for Listeners class

Signed-off-by: VesnaGrumic <vesna.grumic@endava.com>

**Related Issue**
/

**Description of the solution adopted**
/

**Screenshots**
/

**Any side note on the changes made**
/

### CODE REVIEW

**Review by LeoNerdoG — APPROVED** [review]
Hey @Coduz,
I have checked this PR and it looks OK. can you please emrge it after CI finishes? Thanks! 
P.S. Please add label "test" to the PR.

File: client/gateway/provider/fuse/src/test/java/org/eclipse/kapua/client/gateway/mqtt/fuse/FuseChannelTest.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        builder.credentials(Mockito.mock(Credentials.UserAndPassword.class));
+    }
+
+    @Test
+    public void builderBuilderTest() {
+        FuseChannel.Builder builderExpected = new FuseChannel.Builder();
+        assertEquals("Actual and expected values should be the same.", builderExpected, builderExpected.builder());
+    }
+
+    @Test(expected = NoClassDefFoundError.class)
```
Comment (Coduz) [inline client/gateway/provider/fuse/src/test/java/org/eclipse/kapua/client/gateway/mqtt/fuse/FuseChannelTest.java:57]: Same as comments on #3104

File: client/gateway/provider/fuse/src/test/java/org/eclipse/kapua/client/gateway/mqtt/fuse/FuseChannelTest.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    @Test
+    public void handleCloseContextNullTest() throws Exception {
+        FuseChannel fuseChannel = builder.build();
+        fuseChannel.handleClose(null);
+        final CallbackConnection connection = Mockito.mock(CallbackConnection.class);
+        Mockito.verify(connection, Mockito.times(0)).disconnect(null);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void publishMqttTest() throws Exception {
```
Comment (Coduz) [inline client/gateway/provider/fuse/src/test/java/org/eclipse/kapua/client/gateway/mqtt/fuse/FuseChannelTest.java:123]: This is a correct usage of `expected`

**Review by Coduz — CHANGES_REQUESTED** [review]
Hi @VesnaGrumic 

also this PR has the same problems as #3104 

So please fix the usase of `expected` or `assert*`.

Thanks,

_Alberto_

**Pull request closed without merging**